lib/readline.js

```diff
@@ -768,6 +780,49 @@ Interface.prototype._moveCursor = function(dx) {
   }
 };
 
+Interface.prototype._ttyWriteDumb = function(s, key) {
+  key = key || {};
+
+  if (key.name === 'escape') return;
+
+  if (this._sawReturnAt && key.name !== 'enter')
+    this._sawReturnAt = 0;
+
+  if (key.ctrl && key.name === 'c') {
+    if (this.listenerCount('SIGINT') > 0) {
+      this.emit('SIGINT');
+    } else {
+      // This readline instance is finished
+      this.close();
+    }
+  }
+
+  switch (key.name) {
+    case 'return':  // carriage return, i.e. \r
+      this._sawReturnAt = Date.now();
+      this._line();
+      break;
+
+    case 'enter':
+      // When key interval > crlfDelay
+      if (this._sawReturnAt === 0 ||
+          Date.now() - this._sawReturnAt > this.crlfDelay) {
+        this._line();
+      }
+      this._sawReturnAt = 0;
+      break;
+
+    default:
+      if (s instanceof Buffer)
+        s = s.toString('utf-8');
+
+      if (s) {
```

Comment:
This if statement should be obsolete. Even if `s` is indeed an empty string (which would be surprising, the behavior would still stay the same).

Reply:
It could, at least for `escape` key, also there is an example in https://nodejs.org/api/readline.html#readline_rl_write_data_key
That calls .write with first arg null.

```
rl.write(null, { ctrl: true, name: 'u' });
```

Reply:
`null` would not end up in that place in the original function due to being detected as control sequence. That is not the case here but other truthy input is just as faulty and causes issues as well (e.g., `true`). Thus we could just verify that the input is indeed a string (`if (typeof s === 'string) {`) and ignore it otherwise.